Source/JavaScriptCore/b3/B3LowerInt64.cpp

```diff
@@ -385,21 +386,36 @@ class LowerInt64 {
             if (!hasInt64Arg())
                 return;
             Vector<Value*> args;
+            WTF::StdUnorderedSet<size_t> containsLowBitsOfInt64;
             for (size_t index = 1; index < m_value->numChildren(); ++index) {
                 Value* child = m_value->child(index);
                 if (child->type() != Int64)
                     args.append(child);
                 else {
+                    // Offset by one since we add cause we need to pass at least
+                    // one child to the constructor.
+                    containsLowBitsOfInt64.insert(1 + args.size());
                     auto childParts = getMapping(child);
-                    args.append(valueHiLo(childParts.first, childParts.second));
+                    ASSERT(isARM_THUMB2());
+                    // On ARMv7, AirCCallingConvention allocates low registers
+                    // before high registers, so we need to pass the low bits
+                    // first.
+                    args.append(childParts.second);
+                    args.append(childParts.first);
                 }
             }
-            CCallValue* cCall = insert<CCallValue>(m_index, m_value->type(), m_origin, m_value->child(0));
+            Type returnType = m_value->type();
+            if (returnType == Int64)
+                returnType = m_proc.addTuple({ Int32, Int32 });
+            CCallValue* cCall = insert<CCallValue>(m_index, returnType, m_origin, m_value->child(0));
+            cCall->setArgContainsLowBitsOfInt64(containsLowBitsOfInt64);
             RELEASE_ASSERT(cCall->numChildren() == 1);
             cCall->effects = m_value->effects();
             cCall->appendArgs(args);
```

Comment:
Could we just pass in a dummy value instead as padding?

That is, something like Int32, In64, Int64 would become Int32, Const(0), Int64, Int64

It is slightly wasteful, but seems much simpler and wouldn't affect 64-bit at all

Reply:
Yah I did consider that, but then LowerInt64.cpp would have more intimate knowledge of the ARMv7 ABI to know where to introduce those dummy values. And AFAICT we'd still need to mark those Const values somehow to avoid emitting needless instructions (though I guess the cost would be in the noise).